packages/create-expo-module/src/create-expo-module.ts

```diff
@@ -0,0 +1,271 @@
+  const { stdout } = await spawnAsync('npm', ['view', `${packageName}@${version}`, 'dist.tarball']);
+  return stdout.trim();
+}
+
+/**
+ * Gets the username of currently logged in user. Used as a default in the prompt asking for the module author.
+ */
+async function npmWhoamiAsync(targetDir: string): Promise<string | null> {
+  try {
+    const { stdout } = await spawnAsync('npm', ['whoami'], { cwd: targetDir });
+    return stdout.trim();
+  } catch (e) {
+    return null;
+  }
+}
+
+/**
+ * Downloads the template from NPM registry.
+ */
+async function downloadPackageAsync(targetDir: string): Promise<string> {
+  const tarballUrl = await getNpmTarballUrl('expo-module-template');
+
+  await downloadTarball({
+    url: tarballUrl,
+    dir: targetDir,
+  });
+  return path.join(targetDir, 'package');
+}
+
+/**
+ * Asks the user for some data necessary to render the template.
+ * Some values may already be provided by command options, the prompt is skipped in that case.
+ */
+async function askForSubstitutionDataAsync(
+  targetDir: string,
+  options: CommandOptions
+): Promise<SubstitutionData> {
+  const defaultPackageSlug = path.basename(targetDir);
+  const defaultProjectName = defaultPackageSlug
+    .replace(/^./, (match) => match.toUpperCase())
+    .replace(/\W+(\w)/g, (_, p1) => p1.toUpperCase());
+
+  const { slug } = options.target
+    ? { slug: defaultPackageSlug }
+    : await prompts({
+        type: 'text',
+        name: 'slug',
+        message: 'What is the package slug?',
+        initial: defaultPackageSlug,
+      });
+
+  const { name } = options.name
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'name',
+        message: 'What is the project name?',
+        initial: defaultProjectName,
+      });
+
+  const { description } = options.description
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'description',
+        message: 'How would you describe the module?',
+      });
+
+  const { package: projectPackage } = options.package
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'package',
+        message: 'What is the Android package name?',
+        initial: `expo.modules.${defaultPackageSlug.replace(/\W/g, '').toLowerCase()}`,
+      });
+
+  const { author } = options.author
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'author',
+        message: 'Who is the author?',
+        initial: (await npmWhoamiAsync(targetDir)) ?? '',
+      });
+
+  const { license } = options.license
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'license',
+        message: 'What is the license?',
+        initial: 'MIT',
+      });
+
+  const { repo } = options.repo
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'repo',
+        message: 'What is the repository URL?',
+      });
```

Comment:
I'm not a fun of such repetitive code ❌ 
I prefer static/dynamic config object and some logic iterating over such config:
```ts
const promptsQueries: Array<{ name: string; promptConfig: PromptObject }> = [
  {
    name: "slug",
    promptConfig: {
      type: "text",
      name: "slug",
      message: "What is the package slug?",
      initial: defaultPackageSlug
    }
  },
  {
    name: "name",
    promptConfig: {
      type: "text",
      name: "name",
      message: "What is the project name?",
      initial: defaultProjectName
    }
  },
  {
    name: "description",
    promptConfig: {
      type: "text",
      name: "description",
      message: "How would you describe the module?"
    }
  },

  {
    name: "package",
    promptConfig: {
      type: "text",
      name: "package",
      message: "What is the Android package name?",
      initial: `expo.modules.${defaultPackageSlug
        .replace(/\W/g, "")
        .toLowerCase()}`
    }
  },
  {
    name: "author",
    promptConfig: {
      type: "text",
      name: "author",
      message: "Who is the author?",
      initial: (await npmWhoamiAsync(targetDir)) ?? ""
    }
  },
  {
    name: "license",
    promptConfig: {
      type: "text",
      name: "license",
      message: "What is the license?",
      initial: "MIT"
    }
  },
  {
    name: "repo",
    promptConfig: {
      type: "text",
      name: "repo",
      message: "What is the repository URL?"
    }
  }
];

const accumulator = {};
for (const { name, promptConfig } of promptsQueries) {
  const result = options[name] ? options[name] : await prompts(promptConfig);
  accumulator[name] = result;
}
```

Reply:
Well, it's not really the same thing. We need to scope some values in `project` object because it's what `create-react-native-library` does. We need to keep interoperability for further integration. Because of this, the queries would have to be divided into two parts and accumulated separately — which in my opinion becomes less clear.

Reply:
Ok, I can see the point, but accumulating answers does not limit us from transforming such accumulator into something different later on 🤔 
```ts
return {
  project: {
    slug: accumuator.slug,
    name: accumulator.name,
    version: '0.1.0',
    description: accumulator.description,
    package: projectPackage,
  },
  author: accumulator.author,
  license: accumulator.license,
  repo: accumulator.repo,
};
```
Or you can propose better accumulation technique that would, in some clever way, accumulate the responses under specific nested or top level scopes 😁

Reply:
My general question is: why should some code/data format imply worse coding technique and generate more boilerplate code? We have almost unlimited power in terms of specifying types format in this case, don't we? 🤔